net/fetcher.go

```diff
@@ -18,7 +19,7 @@ type Fetcher interface {
 	// FetchTipSets will only fetch TipSets that evaluate to `false` when passed to `done`,
 	// this includes the provided `ts`. The TipSet that evaluates to true when
 	// passed to `done` will be in the returned slice. The returns slice of TipSets is in Traversal order.
-	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, done func(ts types.TipSet) (bool, error)) ([]types.TipSet, error)
+	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, done func(ts types.TipSet) (bool, error), from peer.ID) ([]types.TipSet, error)
```

Comment:
Non-blocking: I generally feel a callback parameter (i.e. done func here) should go last, so that if it's inlined, the peer.ID is dangling several lines below.